tests/atomics/test_InsertOp.cpp

```diff
@@ -0,0 +1,43 @@
+#include "gtest/gtest.h"
+
+#include "Kokkos_Core.hpp"
+
+#include "kokkos-utils/atomics/InsertOp.hpp"
+
+using execution_space = Kokkos::DefaultExecutionSpace;
+
+/**
+ * @file
+ *
+ * @addtogroup unittests
+ *
+ * **Atomic insertion**
+ *
+ * This group of tests check the behavior of our atomic insert operators.
+ */
+
+namespace Kokkos::utils::tests::atomics
+{
+
+//! @test Check that @ref Kokkos::utils::atomics::InsertMin works as expected.
+TEST(atomics, InsertMin)
+{
+    using view_t = Kokkos::View<int*, execution_space>;
+
+    constexpr size_t trials = 150;
+
+    view_t data("data", 1);
+
+    utils::atomics::InsertMin insert_min {};
+
+    Kokkos::parallel_for(
+        "atomics::InsertMin",
+        Kokkos::RangePolicy<execution_space>(0, trials),
+        KOKKOS_LAMBDA(const typename execution_space::size_type trial)
+        {
+            insert_min.op(data, 0, trials - trial);
+        }
+    );
+}
```

Comment:
```suggestion
TEST(atomics, InsertMin)
{
    using view_t = Kokkos::View<int*, execution_space>;

    const view_t data("data", 2);

    const utils::atomics::InsertMin insert_min {};

    Kokkos::parallel_for(
        "atomics::InsertMin",
        Kokkos::RangePolicy<execution_space>(0, 1),
        KOKKOS_LAMBDA(const typename execution_space::size_type /* */)
        {
            insert_min.op(data, 0, -3);
            insert_min.op(data, 1,  2);
        }
    );
    Kokkos::fence("atomics_InsertMin_Test: fence after parallel_for operation");    

    const auto data_h = Kokkos::create_mirror_view_and_copy(Kokkos::DefaultHostExecutionSpace{}, data);

    ASSERT_EQ(data_h(0), -3);
    ASSERT_EQ(data_h(1),  0);    
}
```
(to test the outcome of the operation)

Reply:
OK for yu ?

Reply:
🟢 !